fastcrypto/src/bls12381/mod.rs

```diff
@@ -682,25 +682,34 @@ impl AggregateAuthenticator for BLS12381AggregateSignature {
         if signatures.len() != pks.len() || signatures.len() != messages.len() {
             return Err(FastCryptoError::InputLengthWrong(signatures.len()));
         }
-        let mut pk_iter = pks.into_iter();
-        for i in 0..signatures.len() {
-            let sig = signatures[i].sig;
-            let result = sig
-                .fast_aggregate_verify(
-                    true,
-                    messages[i],
-                    $dst_string,
-                    &pk_iter
-                        .next()
-                        .unwrap()
-                        .map(|x| &x.pubkey)
-                        .collect::<Vec<_>>()[..],
-                );
-            if result != BLST_ERROR::BLST_SUCCESS {
-                return Err(FastCryptoError::GeneralOpaqueError);
-            }
+
+        if signatures.is_empty() {
+            // verify_multiple_aggregate_signatures fails on empty input, but we accept here.
```

Comment:
We should be very careful with that, ideally we shouldn't return true, but catch empty sigs at the dapp layer. This has been used as an attack vector in the past.